resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventInputImpl.java

```diff
@@ -223,8 +143,55 @@ private void processField(final InboundSseEventImpl.Builder inboundEventBuilder,
       else if ("data".equals(name))
       {
          if (lastFieldWasData)
+         {
             inboundEventBuilder.write(SseConstants.EOL);
-         inboundEventBuilder.write(value);
+         }
+         if (!textLike && escape)
+         {
+            byte[] values = null;
+            try
+            {
+               values = valueString.getBytes("ISO-8859-1");
```

Comment:
Again this weird encoding. The spec https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events says "Event streams are always decoded as UTF-8. There is no way to specify another character encoding." so I think this applies to data fields too.

Reply:
Yes.  UTF-8 encoding applies to data field and text/html like content.  For the client side explicitly wants a byte array with Resteasy created @Stream annotation pair with @Produces(MediaType.APPLICATION_OCTET_STREAM), we can handle it to use  ISO-8859-1 to convert the string to bytes array with 8 bit.  This is special path for Resteasy created @Stream feature for push bytes stream. Resteasy can define the media type for data field with these annotations, so it can know user wants bytes without take all data field as text media type with html UTF-8 encoding by default.